app/code/Magento/InventoryBundleIndexer/Model/ResourceModel/GetAllBundleSourceItemsIds.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryBundleIndexer\Model\ResourceModel;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product\Type as ProductType;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Model\ResourceModel\SourceItem;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+
+/**
+ * Get only bundle source items ids.
+ */
+class GetAllBundleSourceItemsIds
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection
+    ) {
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @return array
+     */
+    public function execute(): array
+    {
+        $select = $this->resourceConnection->getConnection()->select();
+        $select
+            ->from(
+                ['source_item' => $this->resourceConnection->getTableName(SourceItem::TABLE_NAME_SOURCE_ITEM)],
+                [SourceItem::ID_FIELD_NAME]
+            )->joinInner(
+                ['product' => $this->resourceConnection->getTableName('catalog_product_entity')],
+                'source_item.' . SourceItemInterface::SKU . ' = product.' . ProductInterface::SKU,
```

Comment:
Looks like we always will have empty result
Because bundle doesn't have source items and children have different SKU (different with parent product)

Reply:
Will it be implemented soon? Because now it works good, bundle product has its own source item.